src/components/structures/RoomView.js

```diff
@@ -650,11 +662,9 @@ module.exports = React.createClass({
         this._loadMembersIfJoined(room);
         this._calculateRecommendedVersion(room);
         this._updateE2EStatus(room);
-        if (!this.state.permalinkCreator) {
-            const permalinkCreator = new RoomPermalinkCreator(room);
-            permalinkCreator.start();
-            this.setState({permalinkCreator});
-        }
+
+        let creator = this._getPermalinkCreatorForRoom(room);
+        if (!creator.isStarted()) creator.start();
```

Comment:
hm, I'm slightly worried if a search result spans lots of rooms, that we'd create listeners for all of those rooms. As the search view is not live, we're also not really interested in observing the most common servers.Also, as it stands, we'd also not stop listening on any of the room until we go to a different room.

I realize this is sort of opposite to what Dave said, but you could call just `load` and not `start` for the link creators created for the search tiles, and only call start on the one created for the current room... e.g. similar to what you had before :see_no_evil:

Search should really be in it's own component :cry:

Reply:
github ate my comments it seems :(

> Search should really be in it's own component 😢

+10000

> hm, I'm slightly worried if a search result spans lots of rooms, that we'd create listeners for all of those rooms. As the search view is not live, we're also not really interested in observing the most common servers.Also, as it stands, we'd also not stop listening on any of the room until we go to a different room.
> 
> I realize this is sort of opposite to what Dave said, but you could call just load and not start for the link creators created for the search tiles, and only call start on the one created for the current room... e.g. similar to what you had before 🙈

The permalink creator for the primary room should be the only that is started, the others are just created. They weren't being loaded though, which is fixed in https://github.com/matrix-org/matrix-react-sdk/pull/2898/commits/ccf292f05368f63dc4f0f5fd01135224d38f5b97 .